app/controllers/api/v1/houses_controller.rb

```diff
@@ -0,0 +1,154 @@
+class Api::V1::HousesController < ApplicationController
+  # load_and_authorize_resource
+  before_action :set_house, only: %i[show edit update destroy]
+
+  # GET /houses or /houses.json
+  def index
+    @houses = House.all
+    render json: @houses
+  end
+
+  # GET /houses/1 or /houses/1.json
+  def show; end
+
+  # GET /houses/new
+  def new
+    @house = House.new
+  end
+
+  # GET /houses/1/edit
+  def edit; end
+
+  # POST /houses or /houses.json
+  def create
+    @house = House.new(house_params)
+
+    if @house.save
+      render json: { status: 'success', message: 'House created successfully!' }, status: :created
+    else
+      render json: { status: 'failed', errors: @house.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /houses/1 or /houses/1.json
+  def update
+    respond_to do |format|
+      if @house.update(house_params)
+        format.html { redirect_to house_url(@house), notice: 'House was successfully updated.' }
+        format.json { render :show, status: :ok, location: @house }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @house.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /houses/1 or /houses/1.json
+  def destroy
+    @house.destroy
+
+    respond_to do |format|
+      format.html { redirect_to houses_url, notice: 'House was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def unique_cities
+    cities = House.distinct.pluck(:city)
+    render json: cities
+  end
+
+  def unique_houses
+    houses = House.distinct.pluck(:house_name)
+    render json: houses
+  end
+
+  # GET /houses/1 or /houses/1.json
+  swagger_path '/houses/{id}' do
+    operation :get do
+      key :summary, 'Fetch a house by ID'
+      key :tags, ['Houses']
+      parameter do
+        key :name, :id
+        key :in, :path
+        key :description, 'ID of the house'
+        key :required, true
+        key :type, :integer
+      end
+      response 200 do
+        key :description, 'House response'
+        schema do
+          key :type, :object
+          key :$ref, :House
+        end
+      end
+      response 404 do
+        key :description, 'House not found'
+      end
+    end
+  end
+
+  # POST /houses or /houses.json
+  swagger_path '/houses' do
+    operation :post do
+      key :summary, 'Create a new house'
+      key :tags, ['Houses']
+      parameter do
+        key :name, :house
+        key :in, :body
+        key :description, 'House parameters'
+        key :required, true
+        schema do
+          key :$ref, :HouseInput
+        end
+      end
+      response 201 do
+        key :description, 'House created successfully'
+        schema do
+          key :type, :object
+          key :$ref, :House
+        end
```

Comment:
- [ ] This code is causing trouble with Puma and the server:

![image](https://github.com/mercymugambi/backend_house_reservations_rails_app/assets/57421823/47eda21d-07ed-4d5b-814d-e8ee1e03eedb)

Does this swagger code belong to this file?